lib/models/diary.js

```diff
@@ -1,35 +1,61 @@
-const environment = process.env.NODE_ENV || 'development';
-const configuration = require('../../knexfile')[environment];
-const database = require('knex')(configuration);
+const env    = process.env.NODE_ENV || 'development';
+const config = require('../../knexfile')[env];
+const db     = require('knex')(config);
 
+const createDiary = name => {
+  return db.raw(
+    'INSERT INTO diary (name, created_at) VALUES (?, ?)', [name, new Date]
+  );
+};
 
-function createDiary(name){
-  return database.raw(
-    'INSERT INTO diary (name, created_at) VALUES (?, ?)',
-    [name, new Date]
-  )
-}
+const emptyDiarysTable = () => {
+  return db.raw(
+    'TRUNCATE diary RESTART IDENTITY CASCADE'
+  );
+};
 
-function findFirst() {
-  return database.raw('SELECT * FROM diary LIMIT 1')
-}
+const find = id => {
+  return db.raw(
+    'SELECT * FROM diary WHERE id = ?', [id]
+  );
+};
 
-function emptyDiarysTable() {
-  return database.raw('TRUNCATE diary RESTART IDENTITY')
-}
+const findByName = name => {
+  return db.raw(
+    'SELECT * FROM diary WHERE name = ?', [name]
+  );
+};
 
-function find(id){
-  return database.raw('SELECT * FROM diary WHERE id=?', [id])
+const all = () => {
+  return db.raw(
+    'SELECT * FROM diary'
+  );
+};
+
+const findFoodsFor = name => {
+  return db.raw(
+    `SELECT foods.* FROM foods
+    JOIN diary_foods ON foods.id = diary_foods.food_id
+    JOIN diary ON diary.id = diary_foods.meal_id
+    WHERE diary.name = ?`, [name]
+  )
+};
+
+const deleteSelected = id => {
+  return db.raw(`DELETE FROM diary WHERE id = ?`, [id])
 }
 
-function findName(name){
-  return database.raw('SELECT * FROM diary WHERE name=?', [name])
+const updateName = (newName, id) => {
+  return db.raw(`UPDATE diary SET name = ? WHERE id = ? RETURNING name`, [newName, id])
 }
 
 module.exports = {
   create: createDiary,
-  first: findFirst,
+  destroyAll: emptyDiarysTable,
   find: find,
-  findName: findName,
-  destroyAll: emptyDiarysTable
-}
+  delete: deleteSelected,
+  updateName: updateName,
+  // findFoodsFor: findFoodsFor,
+  all: all,
```

Comment:
I'm not exactly sure what you mean by "you can just say it once like: `all` ".

Also is it convention to have the key/value be the same?